test/bwe/MatrixTests.cpp

```diff
@@ -46,9 +53,154 @@ TEST(MatrixTest, choleskyStability)
         {5.5034655381268649E-19, 102615.51002615986, -3.4396659613292905E-20},
         {-3.9095946203353344E-29, -3.4396659613292905E-20, 2.443496637709584E-30}});
 
+    ASSERT_TRUE(math::isSymmetric(m));
+    EXPECT_TRUE(isPositiveSemiDefinite2(m));
+    auto mine = std::abs(math::min(m));
+    auto maxe = std::abs(math::max(m));
+
+    auto mf = 0.001000;
+    double detti = math::det(m);
+    auto d2 = math::det2(m);
+    ASSERT_GE(math::det(m), 0);
     auto L = choleskyDecompositionLL(m);
     auto c = L * transpose(L);
     EXPECT_FALSE(m == c);
+    auto d = m - c;
+    auto nrm = math::norm(m - c);
+    EXPECT_LT(math::norm(m - c), 1.E-10);
     EXPECT_TRUE(math::isValid(L));
     // should check determinant of m - c
-}
\ No newline at end of file
+}
+
+#define EXPECT_ERROR_LT(a, b, relativeError)                                                                           \
```

Comment:
This is a re-implementation of `ASSERT_NEAR ` provided by gtest

https://github.com/google/googletest/blob/main/docs/reference/assertions.md#expect_double_eq-expect_double_eq

Reply:
EXPECT_NEAR is comparing with absolute error. This one compares relative error.
It can be done differently I guess.